src/Simulation/Simulation.jl

```diff
@@ -458,6 +456,57 @@ function _guess_optimal_number_of_threads_for_SOR(gs::NTuple{3, Integer}, max_nt
     return min(nextpow(2, max(cld(n+1, 25), 4)), max_nthreads)
 end
 
+"""
+    Grid(sim::Simulation{T}, contact_id::Int) where T
+
+Determines the grid for a certain contact, constrained by its symmetry.
+    
+## Arguments
+* `sim::Simulation{T}`: [`Simulation`](@ref) for which the grid should be determined.
+* `contact_id::Int` : Contact for which the symmetry (specific for this contact) constrained grid should be determined
+"""
+function Grid(sim::Simulation{T}, contact_id::Int) where T
+    symmetry = sim.symmetry[Symbol(string(contact_id))]
+    intervals = _get_grid_intervals(sim.world, symmetry)
+    Grid(sim, world = World{SolidStateDetectors.world_types(sim.world)...}(intervals...))
+end
+
+"""
+    _get_grid_intervals(world::World{T, 3, Cartesian}, symmetry::MirrorSymmetry{T}) where {T}
+
+Determines the grid intervals for a world with certain symmetry constraint.  
+      
+## Arguments
+* `world::World{T, 3, Cartesian}` : Symmetry unconstrained [`World`](@ref)
+* `symmetry::MirrorSymmetry{T}` : Mirror plane which defines a reflective symmetry
+"""
+function _get_grid_intervals(world::World{T, 3, Cartesian}, symmetry::MirrorSymmetry{T}) where {T}
+    x_interval = symmetry.symmetry_plane.normal[1] == 0 ? world.intervals[1] : 
+        SSDInterval{T, SolidStateDetectors.get_boundary_types(world.intervals[1])[1], :closed,
+        SolidStateDetectors.get_boundary_types(world.intervals[1])[3], :reflecting}(world.intervals[1].left, 
+        symmetry.symmetry_plane.origin[1])
+    y_interval = symmetry.symmetry_plane.normal[2] == 0 ? world.intervals[2] : 
+        SSDInterval{T, SolidStateDetectors.get_boundary_types(world.intervals[2])[1], :closed,
+        SolidStateDetectors.get_boundary_types(world.intervals[2])[3], :reflecting}(world.intervals[2].left, 
+        symmetry.symmetry_plane.origin[2])
+    z_interval = symmetry.symmetry_plane.normal[3] == 0 ? world.intervals[3] : 
+        SSDInterval{T, SolidStateDetectors.get_boundary_types(sim.world.intervals[3])[1], :closed,
+        SolidStateDetectors.get_boundary_types(world.intervals[3])[3], :reflecting}(world.intervals[3].left, 
+        symmetry.symmetry_plane.origin[3])
+    intervals = (x_interval, y_interval, z_interval)
+end
+
+function _get_grid_intervals(world::World{T, 3, Cylindrical}, symmetry::MirrorSymmetry{T}) where {T}
+    r_interval = world.intervals[1]
+    φ_interval = symmetry.symmetry_plane.normal[2] == 0 ? world.intervals[2] : 
```

Comment:
Same thing here: a `CylindricalVector` got converted to a `CartesianVector`.

I expect a bug here (`symmetry.symmetry_plane.normal[2] == 0`) even though I did not check it: 
you are checking if the `y` component of a `CartesianPoint` is 0. If we have a mirror symmetry at `φ = 30°`, the normal vector would point at `CartesianVector{T}(-0.5,0.866,0)`. Even though the second entry of `symmetry.symmetry_plane.normal` is not zero, we still want to adjust the limits and boundary types of the `φ_interval`.